client/my-sites/checkout/checkout/checkout-system-decider.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+/**
+ * Internal Dependencies
+ */
+import CheckoutContainer from './checkout-container';
+import CompositeCheckout from './composite-checkout';
+import config from 'config';
+import { abtest } from 'lib/abtest';
+import { getCurrentUserLocale, getCurrentUserCountryCode } from 'state/current-user/selectors';
+
+// Decide if we should use CompositeCheckout or CheckoutContainer
+export default function CheckoutSystemDecider( {
+	product,
+	purchaseId,
+	selectedFeature,
+	couponCode,
+	isComingFromSignup,
+	plan,
+	selectedSite,
+	reduxStore,
+	redirectTo,
+	upgradeIntent,
+	clearTransaction,
+	cart,
+} ) {
+	const countryCode = useSelector( state => getCurrentUserCountryCode( state ) );
+	const locale = useSelector( state => getCurrentUserLocale( state ) );
+
+	if ( shouldShowCompositeCheckout( cart, countryCode, locale ) ) {
+		return (
+			<CompositeCheckout
+				siteSlug={ selectedSite?.slug }
+				siteId={ selectedSite?.ID }
+				product={ product }
+				purchaseId={ purchaseId }
+				couponCode={ couponCode }
+				redirectTo={ redirectTo }
+				feature={ selectedFeature }
+			/>
+		);
+	}
+
+	return (
+		<CheckoutContainer
+			product={ product }
+			purchaseId={ purchaseId }
+			selectedFeature={ selectedFeature }
+			couponCode={ couponCode }
+			isComingFromSignup={ isComingFromSignup }
+			plan={ plan }
+			selectedSite={ selectedSite }
+			reduxStore={ reduxStore }
+			redirectTo={ redirectTo }
+			upgradeIntent={ upgradeIntent }
+			clearTransaction={ clearTransaction }
+		/>
+	);
+}
+
+function shouldShowCompositeCheckout( cart, countryCode, locale ) {
+	if ( config.isEnabled( 'composite-checkout-wpcom' ) ) {
+		return true;
+	}
+	// Disable for domains in the cart
+	if ( cart?.products?.find( product => product.is_domain_registration ) ) {
```

Comment:
In the interest of being maximally paranoid, is it possible that `product` here won't be an object or will not have `is_domain_registration` defined?

Reply:
I'm all for paranoid code! That said, if `products` exists, my research on the backend says that it must have the `is_domain_registration` property. Of course, the backend could change, but that's true of a ton of other things we rely on as well.

Reply:
Also, if the property doesn't exist, it will just evaluate to false, so there's not a ton of risk.